extraction/src/main/java/io/ssc/trackthetrackers/extraction/hadoop/TrackingGraphJob.java

```diff
@@ -65,30 +76,56 @@ public int run(String[] args) throws Exception {
     return 0;
   }
 
-  static class DistinctifyReducer extends Reducer<IntWritable, IntArrayWritable, IntWritable, IntWritable> {
+  static class DistinctifyReducer extends Reducer<IntWritable, TwoDIntArrayWritable, IntWritable, Text> {
 
+	private final String DELIM = "#";
+	
     @Override
-    protected void reduce(IntWritable trackedHost, Iterable<IntArrayWritable> values, Context ctx)
+    protected void reduce(IntWritable trackedHost, Iterable<TwoDIntArrayWritable> values, Context ctx)
         throws IOException, InterruptedException {
-
-      Set<Integer> trackingHosts = new HashSet<Integer>();
-      for (IntArrayWritable indices : values) {
-        for (int index : indices.values()) {
-          trackingHosts.add(index);
-        }
-      }
-
+    	
+      Map<Integer,String> trackingHosts = new HashMap<Integer,String>();
       int trackedHostIndex = trackedHost.get();
-      trackingHosts.remove(trackedHostIndex);
-
-      for (int trackingHostIndex : trackingHosts) {
-        ctx.write(new IntWritable(trackingHostIndex), new IntWritable(trackedHostIndex));
+      
+      //fill out the trackers and tracking type into a set 
+      for (TwoDArrayWritable tdArray : values) {
+      	  
+    	 Writable [][] array = tdArray.get();    	
+    	 for(int i=0; i< array.length; i++){
```

Comment:
two spaces indentation please and please adhere to the style of the remaining code base